### PR TITLE
[TTS] Add univnet scheduler

### PR DESCRIPTION
**Pull request by ArtyomZemlyak**
# What does this PR do ?

Add scheduler for univnet model.

**Collection**: [TTS]

# Changelog 
- Add `get_warmup_steps` into `UnivNetModel`
- Change `configure_optimizers` in `UnivNetModel` for scheduler support

# Usage
Standart usage of scheduler

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by ArtyomZemlyak** [comment]
@okuchaiev 
Hi!
Can you review?

**Review by XuesongYang — APPROVED** [review]
LGTM. Thanks. Out of my curiosity, after applying these schedulers to univnet, did you observe better performance over the one without explicitly applying schedulers?

**Comment by ArtyomZemlyak** [comment]
@XuesongYang 
Some improvements exists. But need more training tests.

sched - CosineAnnealing with warmup ratio 0.02
Yellow - its finetuning from another language and voice.
image
image
image
image

**Comment by XuesongYang** [comment]
> @XuesongYang Some improvements exists. But need more training tests.
> 
> sched - CosineAnnealing with warmup ratio 0.02 Yellow - its finetuning from another language and voice. image image image image

Thank you for sharing! From your plots, it seems promising by tweaking the scheduler's parameters. 

As a side note: we have shifted the vocoder model to hifigan from univnet. So probably we will not invest to improve univnet. But again, you are welcome to share any practical and better configuration setups of univnet if you achieve better performance. We can then pick it up, train/fine-tune it again, and update our univnet NGC model checkpoints.
